hcloud/instances.go

```diff
@@ -88,15 +88,15 @@ func (i *instances) InstanceTypeByProviderID(providerID string) (string, error)
 	return server.ServerType.Name, nil
 }
 
-func (i *instances) AddSSHKeyToAllInstances(user string, keyData []byte) error {
+func (i *instances) AddSSHKeyToAllInstances(ctx context.Context, user string, keyData []byte) error {
 	return errors.New("not implemented")
 }
 
-func (i *instances) CurrentNodeName(hostname string) (types.NodeName, error) {
+func (i *instances) CurrentNodeName(ctx context.Context, hostname string) (types.NodeName, error) {
 	return types.NodeName(hostname), nil
 }
 
-func (i instances) InstanceExistsByProviderID(providerID string) (exists bool, err error) {
+func (i instances) InstanceExistsByProviderID(ctx context.Context, providerID string) (exists bool, err error) {
```

Comment:
Now that you have a context, shouldn’t you pass it around when calling the hcloud client?

```go
server, _, err = i.client.Server.GetByID(context.Background(), id)
```

(A few lines below.)